src/commands/LettuceMeetVisualizer.ts

```diff
@@ -0,0 +1,292 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, EmbedFieldData, MessageEmbed } from 'discord.js';
+import Command from '../Command';
+import { BotClient } from '../types';
+import axios from 'axios';
+
+// URL for Lettuce Meet GraphQL API, and the query to get the event data.
+const URL = 'https://api.lettucemeet.com/graphql';
+const query = `query EventQuery($id: ID!) {
+  event(id: $id) {
+      ...Event_event...EditEvent_event id
+  }
+}
+fragment EditEvent_event on Event {
+  id title description type pollStartTime pollEndTime 
+  maxScheduledDurationMins pollDates isScheduled start end timeZone updatedAt
+}
+fragment Event_event on Event {
+  id title description type pollStartTime pollEndTime maxScheduledDurationMins 
+  timeZone pollDates start end isScheduled createdAt updatedAt user {
+      id
+  }
+  googleEvents {
+      title start end
+  }
+  pollResponses {
+      id user {
+          __typename...on AnonymousUser {
+              name email
+          }...on User {
+              id name email
+          }...on Node {
+              __isNode: __typename id
+          }
+      }
+      availabilities {
+          start end
+      }
+      event {
+          id
+      }
+  }
+}`;
+const LETTUCE_MEET_INTERVAL = 30;
+
+/**
+ * Data returned by the Lettuce Meet GraphQL API.
+ */
+interface BoardData {
+  data: {
+    event: {
+      id: string;
+      title: string;
+      description: string;
+      type: number;
+      pollStartTime: string;
+      pollEndTime: string;
+      maxScheduledDurationMins: number;
+      timeZone: string;
+      pollDates: string[];
+      isScheduled: boolean;
+      start: string;
+      end: string;
+      createdAt: string;
+      updatedAt: string;
+      user: {
+        id: string;
+      };
+      googleEvents: any[];
+      pollResponses: {
+        id: string;
+        user: {
+          __typename: string;
+          name: string;
+          email: string;
+          id: string;
+        };
+        availabilities: UserAvailability[];
+        event: {
+          id: string;
+        };
+      }[];
+    };
+  };
+}
+
+/**
+ * Availability of a user.
+ */
+interface UserAvailability {
+  start: string;
+  end: string;
+}
+
+interface AvailabilityData {
+  numAvailable: number;
+  peopleAvailable: string[];
+}
+
+
+/**
+ * Visualizes the Lettuce Meet board.
+ */
+export default class LettuceMeetVisualizer extends Command {
+  constructor(client: BotClient) {
+    const definition = new SlashCommandBuilder()
+      .setName('lettucemeetvisualizer')
+      .addStringOption((option) =>
+        option.setName('boardcode').setDescription('Lettuce Meet board code').setRequired(true))
+      .addStringOption((option) =>
+        option.setName('filterby').setDescription('Emails or names of the users to filter by').setRequired(false))
```

Comment:
another nit: I think we can use something besides filterby to describe this. Maybe users?